x-pack/plugins/lens/public/editor_frame_service/editor_frame/suggestion_panel.tsx

```diff
@@ -474,64 +514,92 @@ export function SuggestionPanel({
-              <EuiButtonEmpty
-                data-test-subj="lensSubmitSuggestion"
-                size="xs"
-                iconType="refresh"
-                onClick={() => {
-                  dispatchLens(submitSuggestion());
-                }}
-              >
-                {i18n.translate('xpack.lens.sugegstion.refreshSuggestionLabel', {
-                  defaultMessage: 'Refresh',
-                })}
-              </EuiButtonEmpty>
-            </EuiToolTip>
-          )
-        }
-      >
+  const title = (
+    <EuiTitle size="xxs">
+      <h3>
+        <FormattedMessage
+          id="xpack.lens.editorFrame.suggestionPanelTitle"
+          defaultMessage="Suggestions"
+        />
+      </h3>
+    </EuiTitle>
+  );
+  const accordionWithSuggestionsCount = (
+    <EuiFlexGroup gutterSize="none" alignItems="center" justifyContent="spaceBetween">
+      <EuiFlexItem grow={false}>{title}</EuiFlexItem>
+      <EuiFlexItem grow={false}>
         <div
-          className="lnsSuggestionPanel__suggestions"
-          data-test-subj="lnsSuggestionsPanel"
-          role="list"
-          tabIndex={0}
+          css={css`
+            background-color: ${euiThemeVars.euiColorLightShade};
+            height: ${euiThemeVars.euiSizeL};
+            width: ${euiThemeVars.euiSizeL};
+            text-align: center;
+            border-radius: ${euiThemeVars.euiSizeXS};
+          `}
         >
-          {changesApplied ? renderSuggestionsUI() : renderApplyChangesPrompt()}
+          {suggestions.length + 1}
         </div>
-      </EuiAccordion>
-    </div>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+  const buttonContent = wrapSuggestions ? accordionWithSuggestionsCount : title;
+  return (
+    <EuiAccordion
+      id="lensSuggestionsPanel"
+      buttonProps={{
+        'data-test-subj': 'lensSuggestionsPanelToggleButton',
+        paddingSize: wrapSuggestions ? 'm' : 's',
+      }}
+      className="lnsSuggestionPanel"
+      css={css`
+        padding-bottom: ${wrapSuggestions ? 0 : euiThemeVars.euiSizeS};
+      `}
+      buttonContent={buttonContent}
+      forceState={hideSuggestions ? 'closed' : 'open'}
+      onToggle={toggleSuggestions}
+      extraAction={
+        existsStagedPreview &&
+        !hideSuggestions && (
+          <EuiToolTip
+            content={i18n.translate('xpack.lens.suggestion.refreshSuggestionTooltip', {
+              defaultMessage: 'Refresh the suggestions based on the selected visualization.',
+            })}
+          >
+            <EuiButtonEmpty
+              data-test-subj="lensSubmitSuggestion"
+              size="xs"
+              iconType="refresh"
+              onClick={() => {
+                dispatchLens(submitSuggestion());
+              }}
+            >
+              {i18n.translate('xpack.lens.sugegstion.refreshSuggestionLabel', {
+                defaultMessage: 'Refresh',
+              })}
+            </EuiButtonEmpty>
+          </EuiToolTip>
+        )
+      }
```

Comment:
This may be partially tangential (as it also affects the Lens app), but I wanted to mention it here because it's something that has always bugged me. I've never been a fan of the "Refresh" button that appears when users select one of the available suggestions. I understand we probably included it because Lens have no formal "undo" function, and this "Refresh" button provides a means users to either confirm adoption of the newly selected suggestion or to go back to their previous chart configuration. 

What if instead of showing this confirming "Refresh" button, we assume any newly selected suggestion is not adopted until users either 1) make a configuration change to their visualization or 2) save their visualization in Lens (or apply their changes in the Dashboard inline visualization editor). If one of those occur, it is assumed the suggestion has been adopted by the user and the suggestions can be refreshed. If those don't occur, the user can still return to their previous state by clicking the original suggestion tile. We can also change the "Current visualization" text to the appropriate visualization type with this approach. 

Thoughts? CCing @timductive and @ninoslavmiskovic in case they have any concerns about such a change. That said, if we don't want such a change at this time, can we change the placement of this "Refresh" button so that it appears before the suggestion count, rather than after?

![CleanShot 2023-11-21 at 15 08 55](https://github.com/elastic/kibana/assets/3884767/381015db-2e35-4ab0-8760-b2d11e83dd88)

Reply:
I fixed the order! Let's discuss the other suggestion on a separate topic as it is not related with the current change!